lighty-applications/lighty-rcgnmi-app-aggregator/lighty-rcgnmi-app/src/main/java/io/lighty/applications/rcgnmi/app/Arguments.java

```diff
@@ -20,11 +22,35 @@
             + " (If absent, will look on classpath for it")
     private String loggerPath;
 
+    @Parameter(names = {"-t", "--timeout-in-seconds"}, validateWith = ApplicationTimeoutValidator.class,
+               description = "Application timeout in seconds. "
+            + "This parameter specifies max time which application will wait until a timeout exception will be thrown. "
+            + "Default value is 60. (15 - INT.MAX)")
+    private Integer applicationTimeout = 60;
+
     public String getConfigPath() {
         return configPath;
     }
 
     public String getLoggerPath() {
         return loggerPath;
     }
+
+    public Integer getApplicationTimeout() {
+        return applicationTimeout;
+    }
+
+
+    public static final class ApplicationTimeoutValidator implements IParameterValidator {
+
+        @Override
+        public void validate(String name, String value) throws ParameterException {
+            int intValue = Integer.parseInt(value);
```

Comment:
What if we have null value? .. or something not parsable?

Reply:
I have added a handle for NumberFormatException. I do not know the use-case when the user would enter null here. 
If user try to put:
"-t  " Than will receive exception response "Expected a value after parameter -t"
Any other input is taken as a String.